jersey-servers/src/main/java/com/palantir/remoting3/servers/jersey/RequestLimitingInterceptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.servers.jersey;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.ext.ReaderInterceptor;
+import javax.ws.rs.ext.ReaderInterceptorContext;
+
+/**
+ * Limit requests to 50MB unless the endpoint streams the input.
+ */
+public final class RequestLimitingInterceptor implements ReaderInterceptor {
+
+    private static final int MAX_REQUEST_SIZE_BYTES = 50_000_000;
+
+    private final int maxRequestSize;
+
+    RequestLimitingInterceptor() {
+        this(MAX_REQUEST_SIZE_BYTES);
+    }
+
+    @VisibleForTesting
+    RequestLimitingInterceptor(int maxRequestSize) {
+        this.maxRequestSize = maxRequestSize;
+    }
+
+    @Override
+    public Object aroundReadFrom(ReaderInterceptorContext context) throws IOException, WebApplicationException {
+        LimitingInputStream limitedStream = new LimitingInputStream(context.getInputStream());
+        context.setInputStream(limitedStream);
+
+        Object result = context.proceed();
+
+        // If the input stream has not been limited yet then the endpoint streams the input so do not limit
```

Comment:
something like // if we have not thrown yet, the endpoint streams the input or it has been consumed, so do not limit

Reply:
done